### ISSUE TITLE
Forecast Module

### ISSUE THREAD

**Issue by datwater** [California-Data-Collaborative/RateComparison#45]
For the forecast module, the revenue by tier, fixed vs variable and bill impact should only show the future data, not the past data.  What usage and et data are you using in the future too?  It would be good to be able to select behavior/ external factors to model the impacts and sensitivity.  

-Drew

**Comment by christophertull** [comment]
Thanks Drew! Good ideas and great talking today. Will keep you posted.

**Comment by patwater** [comment]
@datwater are you suggesting that the forecasted piece only show hypothetical?  Note the forecast has the same forecasted usage for both baseline / hypothetical (optimistic assumption though note that's what issue #2 is going to tackle) and then there's just different revenue for the different rates.

<img width="263" alt="screen shot 2017-01-26 at 1 48 34 pm" src="https://cloud.githubusercontent.com/assets/1587811/22351783/64e94fea-e3ce-11e6-8628-3ef5e18aa71f.png">

**Comment by datwater** [comment]
I'm suggesting we get rid of hypothetical in the past and show both baseline and hypothetical in the forecast with the ability in the scenario planner to dictate behavior by selecting the usage range- additionally would be great to pick other types of scenarios:
highest inefficiency each month
average customer efficiency
For now would be good to start with taking average usage/et etc over the range selected to use by customer in the forecast

**Comment by patwater** [comment]
K sounds good.  To clarify "highest inefficiency each month" would be what precisely?  A toggle that makes the scenario where each customer uses their maximum amount of inefficiency?  Would also appreciate clarification on average customer efficiency

**Comment by datwater** [comment]
Would look at each account by each month and pick the most inefficient the customer was by most usage above budget- use the usage by tier in that scenario.  Average efficiency would take the average usage over budget.  These are just examples.  I think a first step would be to pick the matching time span to what is used in the projection and pick the relevant year ie if you project 9 months, would pick the 9 month time period of usage as a reference point to project.  Happy to discuss if anyone has any other questions on the enhancement.

**Comment by monobina** [comment]
Yes, getting rid of the past tiers in the forecast makes sense. Great suggestions on the possible scenarios! Will be interesting to set up and analyze. @christophertull  lets find some time to chat about this enhancement. Sounds like an exciting project :-)

**Comment by christophertull** [comment]
Good stuff! @anudeepvanjavakam1 had previously expressed interest in working on this issue. Maybe you two can collaborate? @monobina

**Comment by monobina** [comment]
Sounds good @christophertull . Since it looks like there will different pieces of development for this enhancement, collaborating on these should be productive! Happy to chat more about a good strategy to go about this :-)
